packages/gateway-client/src/worker/p2p-client/streams.ts

```diff
@@ -0,0 +1,372 @@
+import { Stream } from '@libp2p/interface-connection';
+import { EventEmitter } from 'events';
+import { logger } from '../logging';
+import { encode, decode, Packet } from '../p2p-fetch/lob-enc';
+
+class Deferred<T> {
+    promise: Promise<T>;
+    resolve!: (value: T) => void;
+    reject!: (reason?: unknown) => void;
+
+    constructor() {
+        this.promise = new Promise((resolve, reject) => {
+            this.resolve = resolve;
+            this.reject = reject;
+        });
+    }
+}
+
+declare type Callback = (err?: Error) => void;
+
+export class RawStream extends EventEmitter {
+    log = logger.getLogger('worker/p2p-client/streams');
+    private readDeferred = new Deferred<Buffer | null>();
+    private writeDeferred = new Deferred<Buffer | null>();
+
+    get isOpen(): boolean {
+        return this.libp2pStream.stat.timeline.close === undefined;
+    }
+
+    get protocol(): string | undefined {
+        return this.libp2pStream.stat.protocol;
+    }
+
+    constructor(
+        private readonly libp2pStream: Stream,
+        private readonly ports?: MessagePort[]
```

Comment:
I don't see us using this property anywhere?

Reply:
yeah that should just be a variable